components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -61,38 +63,40 @@ public class ClientManager {
 
     private static final Log LOG = LogFactory.getLog(ClientManager.class);
     private final CloseableHttpAsyncClient httpAsyncClient;
+    private final CloseableHttpClient httpClient;
 
-    /**
-     * Creates a client manager.
-     *
-     * @throws WebSubAdapterException on errors while creating the http client.
-     */
     public ClientManager() throws WebSubAdapterException {
 
-        PoolingNHttpClientConnectionManager connectionManager;
         try {
-            connectionManager = createPoolingConnectionManager();
-            LOG.debug("Successfully created PoolingNHttpClientConnectionManager.");
+            PoolingNHttpClientConnectionManager asyncConnectionManager =
+                    createPoolingConnectionManager(PoolingNHttpClientConnectionManager.class);
+            RequestConfig config = createRequestConfig();
+
+            // Initialize HttpAsyncClient
+            HttpAsyncClientBuilder httpAsyncClientBuilder = HttpAsyncClients.custom()
+                    .setDefaultRequestConfig(config)
+                    .setConnectionManager(asyncConnectionManager)
+                    .setSSLContext(createSSLContext());
+            httpAsyncClient = httpAsyncClientBuilder.build();
+            httpAsyncClient.start();
+            LOG.debug("HttpAsyncClient started");
```

Comment:
I think we need to add more context on with regard to which this http client is initiated.